lib/vtls/openssl.c

```diff
@@ -3842,10 +3842,15 @@ static CURLcode ossl_connect_step1(struct Curl_cfilter *cf,
     return CURLE_OUT_OF_MEMORY;
 
   BIO_set_data(bio, cf);
+#ifdef HAVE_SSL_SET0_WBIO
   BIO_up_ref(bio);
   SSL_set0_rbio(backend->handle, bio);
   SSL_set0_wbio(backend->handle, bio);
-
+#else
+  /* old style, with some weird backward compat handling,
+   * we prefer the more controlled case above. */
```

Comment:
"old" here seems to be before 1.1.0 (according to https://www.openssl.org/docs/man1.1.1/man3/SSL_set0_rbio.html) - maybe that can be mentioned here as it might be useful at some point

Reply:
Yes, also libressl added it in some of its versions.

Reply:
Rewrote the comment.

Reply:
Strangely, latest LibreSSL (3.7.0) has `SSL_set0_rbio`, but not `SSL_set0_wbio`. (it has this instead: `BIO * SSL_get_wbio(const SSL *s);`)

Reply:
That sounds very cautious of them.